### ISSUE TITLE
Carstream Block!!!!

### ISSUE THREAD

**Issue by nameor123** [thekirankumar/carstream-android-auto#112]
The new version of Android auto is now v3. Carstream is completely blocked!

**Comment by esmith13** [comment]
Agreed. It is definitely blocked a new way. Repackaging the app no longer works. It still shows under audio playback but not under rightmost icon any longer. Tested with 1.8.3 and followed repackaging instructions which worked great before Android Auto be.

**Comment by kaisertgc** [comment]
Downgrade AA version from 3.0 to 2.9

**Comment by mikimall** [comment]
Well..
I nave never updated to 3.0 and it is 2.9 but since yesterday morning it doesn't work anymore. It keeps saying " Sorry, No more videos  available". So, Imply thet  3.0 can't be the problemi! Maybe Karum can say much more :)

**Comment by mikimall** [comment]
Magically it has started to work again just after having written the comment over here. The only thing I did was to go the internal SD , listen to an mp3 and get back to the carstream home. And voilà...everything started working fine.
P.S: anyway, it is 2.9 ...i don't have auto update and checked it out it was 2.9. Do you all confirm that 3.0 isn't working?

**Comment by kaisertgc** [comment]
Somebody claims it works but not proved. https://github.com/thekirankumar/carstream-android-auto/issues/109

**Comment by neshot123** [comment]
@Scutley..... it worked great ... THANK YOU SO MUCH was really bummed when it stopped.

**Comment by catshow2010** [comment]
Hi, sorry for my bad english, the following happens to me. Uninstall AA 3.0, I go back to version 2.9.574923
suggested and so and all restart and I am in my car with carstream blocked and the next screen attached

![image](https://user-images.githubusercontent.com/36753645/36578049-ade12fd8-1839-11e8-84e8-66f30b2df49c.png)

**Comment by lluisgener** [comment]
Same problem here, reverting to 2.9 fixes the problem.

**Comment by catshow2010** [comment]
this situation occurred reverting to version 2.9.It is not the solution

**Comment by lluisgener** [comment]
Not in my case, and as I understand, not in the others.

With 3.0 doesn't work, with 2.9 it worked before, and keeps working after reverting from 3.0.

**Comment by marco69aurelio** [comment]
Yes, with AA2.9 works. Unistall AA3.0 and revert to AA2.9. Once you revert to AA2.9, do not start it in your mobile, otherwise it will upgrade again to AA3.0. 
Go to PlayStore and block automatic updates. Assuming you have already installed fixed Carstream, then go in your car and just plug your mobile with AA2.9 installed. Once AA starts in your car, start Carstream and it should work fine.

**Comment by harpreet13** [comment]
AA older version is forcing auto update when I open the app. Is anybody else facing this issue?

**Comment by rabelson** [comment]
It only forces it on versions < 2.9

**Comment by harpreet13** [comment]
Yes you are right it's working now. I had downloaded 2.8 just to be safe :)

**Comment by kaisertgc** [comment]
I think google team is watching us)))

**Comment by catshow2010** [comment]
ook this. I revert to 2.9 .I have blocked the update the playstore reinstalled carstream and there is no solution. Surely the developer will solve the issue in a short time
https://user-images.githubusercontent.com/36753645/36578049-ade12fd8-1839-11e8-84e8-66f30b2df49c.png

**Comment by rabelson** [comment]
You're looking in the wrong spot. I have this same behavior. There is that blank audio player in the music section of android auto, but what you're looking for is the part of android auto that is on the car dashboard in the very bottom right of the navigation bar. In my Audi, it'll usually just have the option "return to audi" but with carstream it'll have the carstream app listed there as well. At this point you can use the app, but it does not show if the phone is not plugged in to the car's screen.

**Comment by vicbrilo** [comment]
I worked to go back to version 2.9 of android auto, thank you very much catshow2010

**Comment by vicbrilo** [comment]
Me funciona perfectamente con 2.9

**Comment by Ranvirnegi** [comment]
Hi I have downloaded new version of carstream and used app cloner also but  it is not working. Pls help

**Comment by shmykelsa** [comment]
See issue #116

**Comment by youniluth** [comment]
Reverting to 2.9 works with s6 (AA3) but not with A3 16 (AA 3). The last doesn't show carstream under "return to sync 3"...

**Comment by Tonyms2249** [comment]
Iv done it all and still can't get it to work. Down grade to aa2.9. Tried the apk editor trick. Tried the plug into windows and download from the computer trick. Nothing works.

**Comment by esmith13** [comment]
You need to uninstall AA. Then install AA2.9. Then immediately go in to Google play and in the listing for AA hit the three dots in the upper right and disable auto update for AA. Then install CS1.8.3. 

I just did all of these steps two days ago and it worked fine.

**Comment by Tonyms2249** [comment]
What's cs18.3?

**Comment by esmith13** [comment]
Sorry. CS1.8.3 as in CarStream 1.8.3

**Comment by Tonyms2249** [comment]
So I figured out what's going on. I have the carstream 1.8 downloaded but
when I do the Android auto 2.9 download when I open it it forces me to do
the update now I've already gone into Google play settings and turned off
auto updates. When I download it it tells me it's version 2.9 when I plug
it into the cord into the vehicle and try to start it, it says I must
update Android auto to the latest software version in order to use it. I'm
using a Samsung Note 8.

**Comment by Tonyms2249** [comment]
It would not allow me to email everything all in one e-mail I'm sorry to
send you all the different emails with the attachments. As you can see all
apps are running diversion they should and as you can see it does not work

**Comment by Davidwongym** [comment]
I just wondering when the new version of CarStream will out upload for us? V1.8.4

**Comment by Trvious** [comment]
Im a network engineer and part time android dev. This does NOT work on my 2018 Chevy. Looks like the Google Play services update has blocked it. If you set your Android auto view to developer mode when you go to the last icon on your dashboard you can see all of the apps that it shows as compatible but carplay is not listed it only shows up under the headphones icon which is used for Google Play Music

**Comment by cupral** [comment]
This worked for me (root required):
https://forum.xda-developers.com/android/help/delete-disable-downgrade-prevent-update-t3091832

**Comment by AndreDylan** [comment]
Ho Guys sale problem even downloading the apk 2.9 To be able to access the developer mode of the Android auto app, as in v3. Is not accessible.
I got now promt please update your app,but I cannot cancel so you are stuck.
I also deactivate auto update Google apps in app store...

Is there any code we can build to avoid this issue or create a version where the dev.mode is activated 

My final aim is to be able to mirror my mobile to the screen of Mercedes Benz screen that is not touch screen

**Comment by shmykelsa** [comment]
@AndreDylan you should root your phone and use [AA Phenotype Patcher](https://github.com/Eselter/AA-Phenotype-Patcher)

**Comment by AndreDylan** [comment]
@shmykelsa it is not working. The process assumed the AA had the dev mode locked. The issue here is that the new version does not have at all the dev mode...it is not there.
Plus even downloading  2.9 version and deactivating in Google market the auto update..it still block the app to proceed installation asking you..update the up, the cancel buttom is unresponsive 

Is there any discussion on the mirroring of your phone on the screen of your no touch screen of a car..in my case a Mercedes

**Comment by shmykelsa** [comment]
Developer options Is available in every version of Android Auto.

You simply can't use Android Auto 2.9 because newer Google Play Services block you from doing so. If you want to try it "the hard way" you should downgrade to something like Google Play Services 14 and Android Auto 2.9 and totally disconnect from the internet. If your phone has a recent software, chances are you won't be able to downgrade to Google Play Services 14.

**Comment by AndreDylan** [comment]
Hi again, can you show me how to view the dev mode. When I download the last version the dev.mode does not exist anywhere
The ten times click on about it is not working and it is not executable it seems all the instruction our there are for older versions of AA.

I cannot attach the pic here otherwise I would show you that when you click about and the three dots there are only two option available no dev mode

**Comment by AndreDylan** [comment]
Aww I finally find  how to do it...you need to tap in the top bar not the about as in the older versions thanks
